infra/gp-identity-provider/templates/ldap-ocp-oauth-patch.yaml

```diff
@@ -0,0 +1,42 @@
+{{- if .Values.ldap.enable }}
+apiVersion: redhatcop.redhat.io/v1alpha1
+kind: Patch
```

Comment:
TODO: prüfen ob 2 patches auf die gleiche Resource funktionieren.

Reply:
Alternative: 
den bestehenden Patch so anpassen, dass mehrere patch-ids in demselben Patch sind. Dann sollte man sich auch die mittlerweile verwirrenden if-clauses sparen. 

```apiVersion: redhatcop.redhat.io/v1alpha1
kind: Patch
metadata:
  name: example
  namespace: default
spec:
  patches:
    patch-identity-prov-gepaplexx:
      patchTemplate: |
        spec:
          identityProviders:
            - name: Gepaplexx
              mappingMethod: claim
              type: GitHub
              github:
                clientID: "{{ (index . 1).data.clientId | b64dec }}"
                organizations: {{ (index . 1).data.restrOrgs | b64dec }}
                clientSecret:
                  name: github-oauth-secret
      patchType: application/merge-patch+json
      sourceObjectRefs:
        - apiVersion: v1
          kind: Secret
          name: github-oauth-secret
          namespace: openshift-config
      targetObjectRef:
        apiVersion: config.openshift.io/v1
        kind: OAuth
        name: cluster
        namespace: gp-infrastructure
    patch-identity-prov-gepardec:
      patchTemplate: |
        spec:
          identityProviders:
            - name: Gepardec
              mappingMethod: claim
              type: Google
              google:
                clientID: "{{ (index . 1).data.clientId | b64dec }}"
                hostedDomain: "{{ (index . 1).data.restrDomain | b64dec }}"
                clientSecret:
                  name: google-oauth-secret
      sourceObjectRefs:
        - apiVersion: v1
          kind: Secret
          name: google-oauth-secret
          namespace: openshift-config
      targetObjectRef:
        apiVersion: config.openshift.io/v1
        kind: OAuth
        name: cluster
        namespace: gp-infrastructure
  serviceAccountRef:
    name: default

```

Reply:
entsprechend auch den zusätzlichen Patch löschen. Das bestehende ymal File anpassen. 
Die gesamten Patch-IDs unter if stellen, wenn sie konfiguriert sind. Und um die LDAP Integration erweitern.